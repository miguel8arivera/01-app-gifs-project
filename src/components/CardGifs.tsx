type Props = {
  id?: string;
  title: string;
  url: string;
};

const CardGifs = ({ title, url }: Props) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

export default CardGifs;
