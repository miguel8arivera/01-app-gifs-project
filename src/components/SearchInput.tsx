import { useState } from 'react';

type SearchInputProps = {
  setCategories?: React.Dispatch<React.SetStateAction<string[]>>;
  onNewCategory: (newCategory: string) => void;
};

function SearchInput({ onNewCategory }: SearchInputProps) {
  const [inputSearchValue, setInputSearchValue] = useState<string>('');

  const handleChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = e.target;
    setInputSearchValue(value);
  };

  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (inputSearchValue.trim().length <= 2) return;
    // setCategories && setCategories((prev) => [inputSearchValue, ...prev]);
    onNewCategory && onNewCategory(inputSearchValue.trim());

    setInputSearchValue('');
  };

  return (
    <form className="form" onSubmit={onSubmit}>
      <input
        className="search-input"
        type="search"
        value={inputSearchValue}
        onChange={handleChangeInput}
      />
      <button className="search-button">Search</button>
    </form>
  );
}

export default SearchInput;
