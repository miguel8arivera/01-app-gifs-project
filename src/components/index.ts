export { default as CardGifs } from './CardGifs';
export { default as SearchInput } from './SearchInput';
export { default as TableImage } from './TableImage';
