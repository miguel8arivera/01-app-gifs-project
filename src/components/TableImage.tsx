import CardGifs from './CardGifs';
import useFetchGifs from '../hooks/useFetchGifs';

function TableImage({ category }: { category: string }) {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {images?.map((image) => (
        <CardGifs key={image.id} {...image} />
      ))}
    </>
  );
}

export default TableImage;
