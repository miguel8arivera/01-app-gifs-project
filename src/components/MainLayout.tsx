import SearchInput from './SearchInput';
import TableImage from './TableImage';

type MainLayoutProps = {
  setCategories: React.Dispatch<React.SetStateAction<string[]>>;
  categories: string[];
};

const MainLayout = ({ categories, setCategories }: MainLayoutProps) => {
  return (
    <div>
      <SearchInput setCategories={setCategories} />
      <h3>Aqui se cargara todo el proyecto</h3>
      <TableImage categories={categories} />
    </div>
  );
};

export default MainLayout;
