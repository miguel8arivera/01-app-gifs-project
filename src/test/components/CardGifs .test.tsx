import { render, screen } from '@testing-library/react';
import { CardGifs } from '../../components';

describe('Pruebas en <CardGifs />', () => {
  const title = 'Saytama';
  const url = 'https://localhost/algo.jpg';

  test('debe hacer match con el snaptshot', () => {
    const { container } = render(<CardGifs title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('debe mostrar la imagen con el url y el alt indicados', () => {
    render(<CardGifs title={title} url={url} />);
    // screen.debug();
    // console.log(screen.getByRole('img').alt);
    // expect(screen.getByRole('img').src).toBe(url);

    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('debe mostrar el prop title en el componente como texto', () => {
    render(<CardGifs title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
