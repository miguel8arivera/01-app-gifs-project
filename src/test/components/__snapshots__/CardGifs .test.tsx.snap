// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CardGifs /> debe hacer match con el snaptshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saytama"
      src="https://localhost/algo.jpg"
    />
    <p>
      Saytama
    </p>
  </div>
</div>
`;
